boards/CMakeLists.txt

```diff
@@ -12,3 +12,125 @@ if(EXISTS ${BOARD_DIR}/CMakeLists.txt)
 
   add_subdirectory(${BOARD_DIR} ${build_dir})
 endif()
+
+
+if(CONFIG_BUILD_WITH_TFM)
+  # Set default image versions if not defined elsewhere
+  if (NOT DEFINED TFM_IMAGE_VERSION_S)
+    set(TFM_IMAGE_VERSION_S 0.0.0+0)
+  endif()
+
+  if (NOT DEFINED TFM_IMAGE_VERSION_NS)
+    set(TFM_IMAGE_VERSION_NS 0.0.0+0)
+  endif()
+
+  if (NOT CONFIG_TFM_BL2_FALSE)
+    set(PREPROCESSED_FILE_S "${CMAKE_BINARY_DIR}/tfm/bl2/ext/mcuboot/CMakeFiles/signing_layout_s.dir/signing_layout_s.o")
+    set(PREPROCESSED_FILE_NS "${CMAKE_BINARY_DIR}/tfm/bl2/ext/mcuboot/CMakeFiles/signing_layout_ns.dir/signing_layout_ns.o")
+    set(TFM_MCUBOOT_DIR "${ZEPHYR_TFM_MODULE_DIR}/trusted-firmware-m/bl2/ext/mcuboot")
+  endif()
+
+  # Configure which format (full or hash) to include the public key in
+  # the image manifest
+  if(NOT DEFINED TFM_PUBLIC_KEY_FORMAT)
+    set(TFM_PUBLIC_KEY_FORMAT "full")
+  endif()
+
+  # Set srec_cat binary name
+  find_program(SREC_CAT srec_cat)
+  if(${SREC_CAT} STREQUAL SREC_CAT-NOTFOUND)
+      message(FATAL_ERROR "'srec_cat' not found. Please install it, or add it to $PATH.")
+  endif()
+
+  if (CONFIG_TFM_BL2_FALSE)
+    # Merge tfm_s and zephyr (NS) image to a single binary.
+    set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
+      COMMAND ${SREC_CAT}
+      ARGS $<TARGET_PROPERTY:tfm,TFM_S_HEX_FILE> -intel
+        ${CMAKE_BINARY_DIR}/zephyr/${KERNEL_HEX_NAME} -intel
+        -o ${CMAKE_BINARY_DIR}/tfm_merged.hex -intel
+    )
+
+    set_property(GLOBAL APPEND PROPERTY extra_post_build_byproducts
+      ${CMAKE_BINARY_DIR}/tfm_merged.hex
+    )
+
+  elseif(CONFIG_TFM_MCUBOOT_IMAGE_NUMBER STREQUAL "1")
+    set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
+      COMMAND ${SREC_CAT}
+      ARGS $<TARGET_PROPERTY:tfm,TFM_S_HEX_FILE> -intel
+        ${CMAKE_BINARY_DIR}/zephyr/${KERNEL_HEX_NAME} -intel
+        -o ${CMAKE_BINARY_DIR}/tfm_s_zephyr_ns.hex -intel
+
+      #Sign concatenated binary image with default public key in mcuboot folder
+      COMMAND ${PYTHON_EXECUTABLE} ${TFM_MCUBOOT_DIR}/scripts/wrapper/wrapper.py
+        --layout ${PREPROCESSED_FILE_NS}
+        -k ${CONFIG_TFM_KEY_FILE_NS}
+        --public-key-format ${TFM_PUBLIC_KEY_FORMAT}
+        --align 1
+        -v ${TFM_IMAGE_VERSION_NS}
+        --pad
+        --pad-header
+        ${ADD_NS_IMAGE_MIN_VER}
+        -s auto
+        -H 0x400
```

Comment:
Size of header? Should we have a comment for these arguments?

Reply:
Used Kconfig var instead.